src/docs/design-patterns/cqrs-pattern.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "CQRS - Command Query Responsibility Segregation"
+date: "2023-08-11"
+description: CQRS is a design pattern used in software architecture to address the complexity and performance issues that can arise in systems handling both read (query) and write (command) operations.
+featuredImage: "./images/cqrs-pattern.png"
+---
+
+**CQRS** - Command Query Responsibility Segregation - is a design pattern used in software architecture to address the complexity and performance issues that can arise in systems handling both read (**query**) and write (**command**) operations. CQRS suggests segregating the data models and operations for reads and writes into separate components, optimizing each for its specific use case.
+
+In a traditional monolithic architecture, the same data model and database schema are often used for both reading and writing data. However, as an application grows in complexity, the read and write patterns might have different requirements and performance characteristics. CQRS helps to tackle this by decoupling the data storage and retrieval mechanisms.
+
+## How CQRS Works
+
+These are some key features of CQRS and how it works.
+
+### Separation of Models
+
+Instead of using a single data model for both reading and writing, CQRS recommends having separate models. The command model is optimized for handling write operations, ensuring data consistency and integrity. The query model is designed for efficient data retrieval and is optimized for read operations.
+
+### Command and Query Paths
+
+The application's code is divided into two distinct paths: the Command Path and the Query Path. The Command Path handles write operations, such as creating, updating, or deleting data. The Query Path handles read operations and is optimized for querying and displaying data to users.
+
+![CQRS flowchart - the Query Path represents the user getting data from a read data store. The Command Path represents the user changing data in a writeable data store.](images/cqrs-flow.png)
+
+### Event Sourcing
```

Comment:
ES is probably worth mentioning but this could use some kind of introduction, like:

"Event Sourcing is often combined with CQRS, though it's certainly not a prerequisite for using CQRS in your application."

Reply:
Moved this to a section of common patterns seen with CQRS